tests/engagement/test_churn.py

```diff
@@ -383,3 +384,68 @@ def case(name, expect, desktop, mobile, configured):
 
     for test in test_func:
         test(df)
+
+
+def test_attribution_from_new_profile(effective_version,
+                                      generate_main_summary_data,
+                                      generate_new_profile_data):
+    main_summary = generate_main_summary_data([
+        {'client_id': '1', 'attribution': {'source': 'mozilla.org'}},
+        {'client_id': '3', 'attribution': None},
+        {'client_id': '4', 'attribution': None},
+    ])
+
+    def update_attribution(attribution):
+        # only updates the attribution section in the environment
+        env = copy.deepcopy(data.new_profile_sample['environment'])
+        env['settings']['attribution'] = attribution
+        return env
+
+    # The case for NULL in new_profile and NOT NULL in main_summary is
+    # explicitly not tested here because it is not a well defined behavior.
+    new_profile = generate_new_profile_data([
+        {'client_id': '2', 'environment': update_attribution({'source': 'mozilla.org'})},
+        # overwrite the null value
+        {'client_id': '3', 'environment': update_attribution({'source': 'mozilla.org'})},
+        # overwrite the null value, but is not a ping of itself
```

Comment:
What does this mean?

Reply:
Poorly written, it's the case where the new-profile is used to recover attribution but is outside of the current retention period. I make this distinction because I use new-profile pings to recover clients that don't appear again.